pallets/ethereum-transactions/src/ethereum_transaction.rs

```diff
@@ -0,0 +1,411 @@
+                    name: String::from("_targetT2PublicKey"),
+                    kind: ParamType::FixedBytes(32),
+                }],
+                outputs: Vec::<Param>::new(),
+                constant: false,
+            },
+            call_values: vec![Token::FixedBytes(self.t2_public_key.to_vec())],
+        }
+    }
+}
+
+#[derive(Encode, Decode, Default, Clone, PartialEq, Debug, Eq, MaxEncodedLen, TypeInfo)]
+pub struct SlashValidatorData {
+    pub t2_public_key: [u8; 32],
+}
+
+impl SlashValidatorData {
+    pub fn new(t2_public_key: [u8; 32]) -> SlashValidatorData {
+        SlashValidatorData { t2_public_key }
+    }
+
+    pub fn to_abi(&self) -> EthTransactionDescription {
+        EthTransactionDescription {
+            function_call: Function {
+                name: String::from("slashValidator"),
+                inputs: vec![Param {
+                    name: String::from("_targetT2PublicKey"),
+                    kind: ParamType::FixedBytes(32),
+                }],
+                outputs: Vec::<Param>::new(),
+                constant: false,
+            },
+            call_values: vec![Token::FixedBytes(self.t2_public_key.to_vec())],
+        }
+    }
+}
+
+#[derive(Encode, Decode, Default, Clone, PartialEq, Debug, Eq, MaxEncodedLen, TypeInfo)]
+pub struct ActivateValidatorData {
+    pub t2_public_key: [u8; 32],
+}
+
+impl ActivateValidatorData {
+    pub fn new(t2_public_key: [u8; 32]) -> ActivateValidatorData {
+        ActivateValidatorData { t2_public_key }
+    }
+
+    pub fn to_abi(&self) -> EthTransactionDescription {
+        EthTransactionDescription {
+            function_call: Function {
+                name: String::from("activateValidator"),
+                inputs: vec![Param {
+                    name: String::from("_targetT2PublicKey"),
+                    kind: ParamType::FixedBytes(32),
+                }],
+                outputs: Vec::<Param>::new(),
+                constant: false,
+            },
+            call_values: vec![Token::FixedBytes(self.t2_public_key.to_vec())],
+        }
+    }
+}
+
+pub type TransactionId = u64;
+pub type EthereumTransactionHash = H256;
+
+#[derive(Encode, Decode, Clone, PartialEq, Eq, Debug, Default, TypeInfo)]
+pub struct EthTransactionCandidate {
+    pub tx_id: TransactionId,
+    pub from: Option<[u8; 32]>, // AvN Public Key of a validator
+    pub call_data: EthTransactionType,
+    pub signatures: EthSignatures,
+    pub quorum: u32, // number of signatures needed
+
+    // TODO [TYPE: business logic][PRI: high] this needs review since we allow locks to expire, it
+    // is possible to send the same transaction twice and then we need to update this value.
+    // Example: we create a lock to send a transaction to Ethereum. We send it and receive a
+    // temporary `eth_tx_hash` The transaction later becomes stale and has to be resent.
+    // But because tx_hash can only be set once, we will not be able to update it as is.
```

Comment:
```suggestion
    // TODO [TYPE: business logic][PRI: high] this needs review since we allow locks to expire, it
    // is possible to send the same transaction twice and then we need to update this value.
    // Example: we create a lock to send a transaction to Ethereum. We send it and receive a
    // temporary `eth_tx_hash`. The transaction later becomes stale and has to be resent.
    // But because tx_hash can only be set once, we will not be able to update it as is.
```